backend/app/Observers/UserRelationObserver.php

```diff
@@ -0,0 +1,28 @@
+<?php
+
+namespace App\Observers;
+
+use App\Models\UserNotification;
+use App\Models\UserRelation;
+
+class UserRelationObserver
+{
+    public $afterCommit = true;
+
+    public function created(UserRelation $userRelation)
+    {
+        UserNotification::create([
+            'user_id' => $userRelation->following_id,
+            'notifiable_type' => 'App\Models\UserRelation',
+            'notifiable_id' => $userRelation->id,
+        ]);
+    }
+
+    public function deleted(UserRelation $userRelation)
+    {
+        UserNotification::where([
+            'notifiable_type' => 'App\Models\UserRelation',
+            'notifiable_id' => $userRelation->id,
+        ])->first()->delete();
```

Comment:
```suggestion
        ])->delete();
```

If your query is deleting exactly 1 record you can remove the first , but if you are querying multiple records and deleting the first one then retain